### ISSUE TITLE
Dropdown element is hidden by next row

### ISSUE THREAD

**Issue by Zerych** [swimlane/ngx-datatable#847]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
If you add a dropdown element inside row and expand it, it will be covered by the next row. I'm using scrollV in my table.

**Expected behavior**
Dropdown hovers next row and is visible that way.

* **Table version:**
latest

* **Angular version:** 2.0.x
4.0.0

**Comment by Zerych** [comment]
![screenshot from 2017-07-04 17-14-26](https://user-images.githubusercontent.com/16525613/27833631-511c8eba-60dc-11e7-86c2-f6dd529925ea.png)

**Comment by bheda91** [comment]
Having the same issue. 

@amcdnl Please suggest any solution can be done using CSS or some tricks.

**Comment by Zerych** [comment]
@bheda91 I've fixed it by reverting all rows' z-indexes, if you're interested. But that's a terrible solution.

`let rows = document.getElementsByClassName('datatable-row-wrapper');
      for (let i = 0; i < rows.length; i++) {
        let row = <HTMLElement>rows[i];
        row.style.zIndex = '' + (1000 - i);
      }`

**Comment by bheda91** [comment]
Thanks @Zerych, your solution will work. But terrible one. :(

Can be do some better solution like, suggested in this https://github.com/angular-ui/ui-select/issues/308

**Comment by PankajAS** [comment]
@Zerych z-index not working

**Comment by Ashwin2488** [comment]
@Zerych @bheda91 Will setting the Z-index of currently selected row to higher value work? 
 ```
.datatable-row-wrapper:hover {
  z-index: 12; // higher than other rows
}
```

**Comment by lmedeiros** [comment]
Try adding this css style

.datatable-body-cell {
overflow-x: visible !important;
}

It worked for me with a bootstrap dropdown inside a cell

**Comment by vincent-cm** [comment]
To modify css z-index and overflow properties won't work, because this issue is caused by the cascaded DOM order. The correct solution is to modify the source HTML template, DOM structure.

**Comment by DmitriyIvanko** [comment]
I found this solution (may help somebody):
1) use @ng-select/ng-select
2) and add attribute: appendTo="body", example:

        <ng-select [items]="simpleItems"
                   appendTo="body"
                   [(ngModel)]="selectedSimpleItem">
        </ng-select>

No need any other z-index configurations.

**Comment by vincent-cm** [comment]
@DmitriyIvanko Yes you're right, the plugin helped to append the selectable outside the parent div, programmatically.

**Comment by KateFr** [comment]
https://valor-software.com/ngx-bootstrap/#/dropdowns#container

solved problem of in row  dropdown for me

**Comment by idzeravianka** [comment]
if you don't need virtual scroll, disable it, and problem will be fixed

**Comment by melanialvarez** [comment]
For me it worked:

.datatable-row-wrapper {
  z-index: 1; // Set an smaller value for all rows

  &:hover {
    z-index: 2; // increase the value when hovering
  }
}
